### PR TITLE
Fixup pie/sunburst unhover post plot/calc restyle calls

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/3618 and another (pretty bad) pie bug, to be merged into the sunburst PR in https://github.com/plotly/plotly.js/pull/3594

Updated jsbin: https://output.jsbin.com/lesidaxoru/1

cc @plotly/plotly_js

### CODE REVIEW

File: src/traces/pie/plot.js [hunk 1]
```diff
@@ -313,12 +313,12 @@ function attachFxHandlers(sliceTop, gd, cd) {
 
     // hover state vars
     // have we drawn a hover label, so it should be cleared later
-    var hasHoverLabel = false;
+    if(!('_hasHoverLabel' in trace)) trace._hasHoverLabel = false;
```
Comment (archmoj) [inline src/traces/pie/plot.js]: What about using the line below?
<pre>trace._hasHoverLabel = !!trace._hasHoverLabel;</pre>

Comment (etpinard) [reply src/traces/pie/plot.js]: I don't want to cast `undefined` to `false`.

File: src/traces/pie/base_plot.js [hunk 2]
```diff
@@ -16,8 +16,7 @@ exports.name = 'pie';
 exports.plot = function(gd) {
     var Pie = Registry.getModule('pie');
     var cdPie = getModuleCalcData(gd.calcdata, Pie)[0];
-
-    if(cdPie.length) Pie.plot(gd, cdPie);
+    Pie.plot(gd, cdPie);
```
Comment (archmoj) [inline src/traces/pie/base_plot.js]: @etpinard Could you please elaborate why the condition check is no longer required here?

Comment (etpinard) [reply src/traces/pie/base_plot.js]: Because we need `Pie.plot` to `.exit().remove()` the old pie traces.

**Comment by archmoj** [comment]
@etpinard Thanks.
Looks very good.
:dancer:

**Pull request merged**